geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/FeatureCacheEventTimeOrdering.scala

```diff
@@ -0,0 +1,51 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.index
+
+import java.util.Date
+
+import org.opengis.feature.simple.SimpleFeature
+import org.opengis.filter.Filter
+import org.opengis.filter.expression.Expression
+import org.slf4j.LoggerFactory
+
+class FeatureCacheEventTimeOrdering(baseCache: KafkaFeatureCache, attr: Expression) extends KafkaFeatureCache {
+  private val log = LoggerFactory.getLogger(classOf[FeatureCacheEventTimeOrdering])
+
+  override def put(feature: SimpleFeature): Unit = {
+    baseCache.query(feature.getID) match {
+      case None => baseCache.put(feature)
+
+      case Some(cur) =>
+        val dtg = attr.evaluate(feature).asInstanceOf[Date]
+        val curDate = attr.evaluate(cur).asInstanceOf[Date]
```

Comment:
there are some null checks in the expiry code, do we need to worry about nulls here?